From ae2bae2d93c0fef68c89bd2819bdbbfe6f91010d Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Fri, 1 May 2015 16:11:44 +0530
Subject: [PATCH] pmic-ccsm: Fix the bc1.2 charger detection completion status

When the charger detection failure happens, the detection logic
is not re-trying. So the charger detection completion status to
be validated correctly.

Change-Id: Icceea6b2eae720958ecb263f04ceca9e8b84742e
Tracked-on: https://jira01.devtools.intel.com/browse/IMINAN-34835
Signed-off-by: R, Kannappan <r.kannappan@intel.com>
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
---
 drivers/power/intel_pmic_ccsm.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index 8575093..1c82370 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -854,13 +854,15 @@ static int get_charger_type(void)
 		i++;
 		dev_dbg(chc.dev, "Read USBSRCDETSTATUS val: %x\n", val);
 
-		if (val & USBSRCDET_SUSBHWDET_DETSUCC)
+		if ((val & USBSRCDET_SUSBHWDET_DETSUCC) ==
+				USBSRCDET_SUSBHWDET_DETSUCC)
 			break;
 		else
 			msleep(USBSRCDET_SLEEP_TIME);
 	} while (i < USBSRCDET_RETRY_CNT);
 
-	if (!(val & USBSRCDET_SUSBHWDET_DETSUCC)) {
+	if ((val & USBSRCDET_SUSBHWDET_DETSUCC) !=
+			USBSRCDET_SUSBHWDET_DETSUCC) {
 		dev_err(chc.dev, "Charger detection unsuccessful after %dms\n",
 			i * USBSRCDET_SLEEP_TIME);
 		return 0;
-- 
1.9.1

